File: jenkins-plugin/src/main/java/org/eazyportal/plugin/release/jenkins/action/FinalizeSnapshotVersionActionFactory.java
package org.eazyportal.plugin.release.jenkins.action;

import hudson.Extension;
import hudson.model.InvisibleAction;
import org.eazyportal.plugin.release.core.action.FinalizeSnapshotVersionAction;
import org.eazyportal.plugin.release.core.scm.ScmActions;

import java.io.Serializable;

@Extension
public class FinalizeSnapshotVersionActionFactory extends InvisibleAction implements Serializable {

    public FinalizeSnapshotVersionAction create(ScmActions scmActions) {
        return new FinalizeSnapshotVersionAction(scmActions);
    }

}

File: gradle-plugin/src/main/kotlin/org/eazyportal/plugin/release/gradle/action/FinalizeSnapshotVersionActionFactory.kt
package org.eazyportal.plugin.release.gradle.action

import org.eazyportal.plugin.release.core.action.FinalizeSnapshotVersionAction
import org.eazyportal.plugin.release.gradle.model.EazyReleasePluginExtension

class FinalizeSnapshotVersionActionFactory : ReleaseActionFactory<FinalizeSnapshotVersionAction> {

    override fun create(extension: EazyReleasePluginExtension): FinalizeSnapshotVersionAction =
        FinalizeSnapshotVersionAction(extension.projectActionsFactory, extension.scmActions)

}